### ISSUE TITLE
Use ES6 classes in our JS

### ISSUE THREAD

**Issue by yuvipanda** [jupyterhub/binderhub#1373]
Our JS uses old school prototype based classes. I think it's time we switch to a more modern, cleaner ES6 class syntax. I think it's easier to understand what is happening, and helps newer contributors who might just have modern JS experience.

**Comment by manics** [comment]
I'm in favour of using `prettier` and `black` throughout this repo with pre-commit for autoformatting (code, javascript, markdown, yml). Main thing is to decide when to make the change to avoid conflicts with pending PRs.

I know you've previously said pre-commit is too complicated but it doesn't have to be, just `pre-commit run -a` is all I ever do, I never bother with the automated hooks :smiley: 

I'm happy to make that change along with docs if you want.

**Comment by consideRatio** [comment]
Yeah I've been very very happy with pre-commit since a while back, the experience got even better with pre-commit.ci.

`prettier`, `black`, and `pyupgrade` are amazing!
